### PR TITLE
LPS-187626 clay:tabs taglib is adding additional tabs panel

### PR DESCRIPTION
**Pull request by dsanz**
This supersedes https://github.com/liferay-frontend/liferay-portal/pull/3915. There, 
* We removed an extra `<div>` panel coming from the taglib-generated SSR markup (see [LPS-187626](https://liferay.atlassian.net/browse/LPS-187626) for details), which was redundant when React clay tabs component takes control.
* In doing so we chose to `appendChild()` all panel child nodes (using [firstChild](https://developer.mozilla.org/en-US/docs/Web/API/Node/firstChild) as we do in our [build_fragment utility](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/build_fragment.js#L15-L17), not just `firstElementChild`)
* This way, some artificial styles DXP apps must add to handle this dual markup can be deleted

In addition, this new PR:
* Improves the above by adding the panels into a `documentFragment` to avoid browser reflows that could be caused if child nodes are added to the markup one by one in the active tab
* Manages the `active` value from the DXP Tabs component, so that now taglib works fine when active tab is not the first tab. Before this, there was a flicker while rendering, where first tab was visible for some milliseconds between the SSR and the react rendering.

Note there's still an extra `<div>` produced by the `Panel` component, which proved specially hard to remove from markup. That can be a future improvement if it happens to cause any issue we don't see at this moment.

cc @sandrodw3 @markocikos 

fyi, in case you guys are curious about this part of the frontend architecture @kresimir-coko @chestofwonders @juanjofgliferay

### CODE REVIEW

**Comment by dsanz** [comment]
ci:test:stable

**Comment by dsanz** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js [hunk 1]
```diff
@@ -4,14 +4,20 @@
...
-		ref.current.appendChild(children);
-	}, [children]);
+		const fragment = document.createDocumentFragment();
+
+		while (tabPanel.firstChild) {
+			fragment.appendChild(tabPanel.firstChild);
+		}
+
+		ref.current.appendChild(fragment);
+	}, [tabPanel]);
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js:20]: Why this direct DOM compexity? Can we avoid `useEffect` and just do something like 
```js
return <>{children}</>;
```

**Review by markocikos — COMMENTED** [review]
@dsanz A question inline. Other than that, looks good!

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js (hunk 1 above)
Comment (dsanz) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js:20]: this comes from the original impl of the clay tabs tag. The thing is that `{children}` here is not a list of React.element but just DOM elements representing the children of the tabs as they come from the SSR. We may review this in subsequent tasks as I'm not extremely happy with this duality

Comment (sandrodw3) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js:20]: Yeah, I tried exactly what you suggest @markocikos and found out what @dsanz explained

Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js:20]: Got it. Yeah, this makes sense then, unless we want to refactor upstream code :+1:

**Review by markocikos — APPROVED** [review]
LGTM! :sparkles:

**Comment by sandrodw3** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
No unique failures.

**Comment by Tim-Cao** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/146121

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**